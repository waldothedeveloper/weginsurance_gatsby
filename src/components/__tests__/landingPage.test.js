import { render, screen } from "@testing-library/react";

import LandingPage from "../landingPage";
import React from "react";
import { navigate } from "gatsby";
import userEvent from "@testing-library/user-event";

const user = userEvent.setup();

jest.mock(`gatsby-plugin-image`, () => {
  const React = require(`react`);
  const plugin = jest.requireActual(`gatsby-plugin-image`);

  const mockImage = ({ imgClassName, ...props }) =>
    React.createElement(`img`, { ...props, className: imgClassName });

  const mockPlugin = {
    ...plugin,
    GatsbyImage: jest.fn().mockImplementation(mockImage),
    StaticImage: jest.fn().mockImplementation(mockImage),
  };

  return mockPlugin;
});

describe(`LandingPage`, () => {
  it(`renders correctly`, () => {
    const container = render(<LandingPage />);
    expect(container).toMatchSnapshot();
  });

  it(`navigates to /auth/login when Gatsby Link (anchor) is clicked`, async () => {
    render(<LandingPage />);

    await user.click(screen.getAllByText(/Entrar al sistema/i)[0]);

    await navigate(`/auth/login`);
    expect(navigate).toHaveBeenCalledTimes(1);
    expect(navigate).toHaveBeenCalledWith(`/auth/login`);
  });
});
