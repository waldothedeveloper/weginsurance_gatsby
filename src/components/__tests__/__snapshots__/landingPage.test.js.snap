// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-white pb-8 sm:pb-12 lg:pb-12"
      >
        <div
          class="overflow-hidden pt-8 sm:pt-12 lg:relative lg:py-48"
        >
          <div
            class="mx-auto max-w-md px-4 sm:max-w-3xl sm:px-6 lg:grid lg:max-w-7xl lg:grid-cols-2 lg:gap-24 lg:px-8"
          >
            <div>
              <div>
                <img
                  alt="weginsurance logo"
                  src="../images/weginsurance_logo.jpeg"
                />
              </div>
              <div
                class="mt-20"
              >
                <div>
                  <a
                    class="inline-flex space-x-4"
                    href="#"
                  >
                    <span
                      class="rounded bg-sky-50 px-2.5 py-1 text-xs font-semibold uppercase tracking-wide text-sky-600"
                    >
                      Que hay de nuevo?
                    </span>
                    <span
                      class="inline-flex items-center space-x-1 text-sm font-medium text-sky-600"
                    >
                      <span>
                        Ahora puedes enviar fotos por mensajes
                      </span>
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </a>
                </div>
                <div
                  class="mt-6 sm:max-w-xl"
                >
                  <h1
                    class="text-4xl font-extrabold tracking-tight text-gray-900 sm:text-5xl"
                  >
                    Bienvenido al sistema de Mensajes
                  </h1>
                  <p
                    class="mt-6 text-xl text-gray-500"
                  >
                    Si tiene problemas para entrar al sistema contacte al soporte tecnico.
                  </p>
                </div>
                <div
                  class="mt-12 sm:flex sm:w-full sm:max-w-lg"
                >
                  <div
                    class="mt-4 sm:mt-0 sm:ml-3"
                  >
                    <a
                      class="block max-w-fit rounded-md border border-transparent bg-sky-600 px-5 py-3 text-base font-medium text-white shadow hover:bg-sky-700 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2 sm:px-10 md:max-w-md"
                      data-testid="login-button"
                      href="/auth/login"
                    >
                      Entrar al sistema
                    </a>
                  </div>
                </div>
                <div
                  class="mt-6"
                >
                  <div
                    class="inline-flex items-center divide-x divide-gray-300"
                  >
                    <div
                      class="flex flex-shrink-0 pr-5"
                    >
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5 text-yellow-400"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5 text-yellow-400"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5 text-yellow-400"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5 text-yellow-400"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5 text-yellow-400"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                        />
                      </svg>
                    </div>
                    <div
                      class="min-w-0 flex-1 py-1 pl-5 text-sm text-gray-500 sm:py-3"
                    >
                      <span
                        class="font-medium text-gray-900"
                      >
                        Si tiene sugerencias acerca de el sistema dejanos saber dando
                      </span>
                       
                      <span
                        class="font-medium text-sky-600"
                      >
                        click aqui
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sm:mx-auto sm:max-w-3xl sm:px-6"
          >
            <div
              class="py-12 sm:relative sm:mt-12 sm:py-16 lg:absolute lg:inset-y-0 lg:right-0 lg:w-1/2"
            >
              <div
                class="hidden sm:block"
              >
                <div
                  class="absolute inset-y-0 left-1/2 w-screen rounded-l-3xl bg-gray-50 lg:left-80 lg:right-0 lg:w-full"
                />
                <svg
                  class="absolute top-8 right-1/2 -mr-3 lg:left-0 lg:m-0"
                  fill="none"
                  height="392"
                  viewBox="0 0 404 392"
                  width="404"
                >
                  <defs>
                    <pattern
                      height="20"
                      id="837c3e70-6c3a-44e6-8854-cc48c737b659"
                      patternUnits="userSpaceOnUse"
                      width="20"
                      x="0"
                      y="0"
                    >
                      <rect
                        class="text-gray-200"
                        fill="currentColor"
                        height="4"
                        width="4"
                        x="0"
                        y="0"
                      />
                    </pattern>
                  </defs>
                  <rect
                    fill="url(#837c3e70-6c3a-44e6-8854-cc48c737b659)"
                    height="392"
                    width="404"
                  />
                </svg>
              </div>
              <div
                class="relative -mr-40 pl-4 sm:mx-auto sm:max-w-3xl sm:px-0 lg:h-full lg:max-w-none lg:pl-12"
              >
                <img
                  alt="girl smiling representing a bright future for the weginsurance's company"
                  layout="fullWidth"
                  placeholder="blurred"
                  src="../images/girl_smiling.avif"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-white pb-8 sm:pb-12 lg:pb-12"
    >
      <div
        class="overflow-hidden pt-8 sm:pt-12 lg:relative lg:py-48"
      >
        <div
          class="mx-auto max-w-md px-4 sm:max-w-3xl sm:px-6 lg:grid lg:max-w-7xl lg:grid-cols-2 lg:gap-24 lg:px-8"
        >
          <div>
            <div>
              <img
                alt="weginsurance logo"
                src="../images/weginsurance_logo.jpeg"
              />
            </div>
            <div
              class="mt-20"
            >
              <div>
                <a
                  class="inline-flex space-x-4"
                  href="#"
                >
                  <span
                    class="rounded bg-sky-50 px-2.5 py-1 text-xs font-semibold uppercase tracking-wide text-sky-600"
                  >
                    Que hay de nuevo?
                  </span>
                  <span
                    class="inline-flex items-center space-x-1 text-sm font-medium text-sky-600"
                  >
                    <span>
                      Ahora puedes enviar fotos por mensajes
                    </span>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </a>
              </div>
              <div
                class="mt-6 sm:max-w-xl"
              >
                <h1
                  class="text-4xl font-extrabold tracking-tight text-gray-900 sm:text-5xl"
                >
                  Bienvenido al sistema de Mensajes
                </h1>
                <p
                  class="mt-6 text-xl text-gray-500"
                >
                  Si tiene problemas para entrar al sistema contacte al soporte tecnico.
                </p>
              </div>
              <div
                class="mt-12 sm:flex sm:w-full sm:max-w-lg"
              >
                <div
                  class="mt-4 sm:mt-0 sm:ml-3"
                >
                  <a
                    class="block max-w-fit rounded-md border border-transparent bg-sky-600 px-5 py-3 text-base font-medium text-white shadow hover:bg-sky-700 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2 sm:px-10 md:max-w-md"
                    data-testid="login-button"
                    href="/auth/login"
                  >
                    Entrar al sistema
                  </a>
                </div>
              </div>
              <div
                class="mt-6"
              >
                <div
                  class="inline-flex items-center divide-x divide-gray-300"
                >
                  <div
                    class="flex flex-shrink-0 pr-5"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5 text-yellow-400"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z"
                      />
                    </svg>
                  </div>
                  <div
                    class="min-w-0 flex-1 py-1 pl-5 text-sm text-gray-500 sm:py-3"
                  >
                    <span
                      class="font-medium text-gray-900"
                    >
                      Si tiene sugerencias acerca de el sistema dejanos saber dando
                    </span>
                     
                    <span
                      class="font-medium text-sky-600"
                    >
                      click aqui
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sm:mx-auto sm:max-w-3xl sm:px-6"
        >
          <div
            class="py-12 sm:relative sm:mt-12 sm:py-16 lg:absolute lg:inset-y-0 lg:right-0 lg:w-1/2"
          >
            <div
              class="hidden sm:block"
            >
              <div
                class="absolute inset-y-0 left-1/2 w-screen rounded-l-3xl bg-gray-50 lg:left-80 lg:right-0 lg:w-full"
              />
              <svg
                class="absolute top-8 right-1/2 -mr-3 lg:left-0 lg:m-0"
                fill="none"
                height="392"
                viewBox="0 0 404 392"
                width="404"
              >
                <defs>
                  <pattern
                    height="20"
                    id="837c3e70-6c3a-44e6-8854-cc48c737b659"
                    patternUnits="userSpaceOnUse"
                    width="20"
                    x="0"
                    y="0"
                  >
                    <rect
                      class="text-gray-200"
                      fill="currentColor"
                      height="4"
                      width="4"
                      x="0"
                      y="0"
                    />
                  </pattern>
                </defs>
                <rect
                  fill="url(#837c3e70-6c3a-44e6-8854-cc48c737b659)"
                  height="392"
                  width="404"
                />
              </svg>
            </div>
            <div
              class="relative -mr-40 pl-4 sm:mx-auto sm:max-w-3xl sm:px-0 lg:h-full lg:max-w-none lg:pl-12"
            >
              <img
                alt="girl smiling representing a bright future for the weginsurance's company"
                layout="fullWidth"
                placeholder="blurred"
                src="../images/girl_smiling.avif"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
